src/uwtools/scheduler.py

```diff
@@ -22,7 +23,8 @@ class JobScheduler(ABC):
     """
 
     def __init__(self, props: dict[str, Any]):
-        self._props = {k: v for k, v in props.items() if k != "scheduler"}
+        self._scheduler = props[STR.scheduler]
```

Comment:
Store the name of the scheduler in a private instance variable so it can be used in an error message below.